### PR TITLE
T23446 tutorial v3

### PR DESCRIPTION
**Pull request by manuq**

### CODE REVIEW

File: examples/tutorial/index.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+const Youtubedl = require('youtube-dl');
+
+Youtubedl.getInfo = util.promisify(Youtubedl.getInfo);
+
+const feedURI = 'https://creativecommons.org/blog/feed/';
+
+const vimeoClientID = '(fill in client ID here)';
+const vimeoClientSecret = '(fill in client secret here)';
+
+const ensureVimeoClient = (function () {
```
Comment (dnohales) [inline examples/tutorial/index.js]: Use arrow function or a regular function.

File: examples/tutorial/index.js [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+        note: { length },
+    };
+}
+
+const hasAncestor = (tagName, scoreIfHas) => fnode => {
+    const lowerTag = tagName.toLowerCase();
+    for (let element = fnode.element, parent;
+        (parent = element.parentNode) !== null &&
+         parent.nodeType === parent.ELEMENT_NODE;
+        element = parent) {
```
Comment (dnohales) [inline examples/tutorial/index.js]: Man, isn't there a way to simplify this `for`? :P

File: examples/tutorial/index.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+            'used under CC BY 4.0 International / Reformatted from original';
+    }
+
+    parseTags ($) {
+        // Wordpress distinguishes predefined "categories" and free-form "tags".
+        // We are likely to make Wordpress categories into featured sets, and
+        // Wordpress tags non-featured. For now, we will mark the tag IDs of
+        // Wordpress tags with "tag:".
+        const wpCategory = $('meta[property="article:section"]')
+              .attr('content');
```
Comment (dnohales) [inline examples/tutorial/index.js]: Too much indentation

File: examples/tutorial/index.js [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        const wpTags = $('meta[property="article:tag"]')
+              .map(function () {
```
Comment (dnohales) [inline examples/tutorial/index.js]: Too much indentation

File: examples/tutorial/index.js [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+        ];
+    }
+
+    processWithFathom ($body) {
+        const _dom = JSDOM.jsdom($body.html(), {
+            features: { ProcessExternalResources: false },
+        });
+        const facts = rules.against(_dom);
+        const html = facts.get('content')
+              .filter(fnode => fnode.scoreFor('paragraphish') > 0)
```
Comment (dnohales) [inline examples/tutorial/index.js]: Too much indentation

File: examples/tutorial/index.js [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+        return { $body: $('article') };
+    }
+
+    async processVimeoEmbeds ($body) {
+        const videoAssets = [];
+
+        // Identify embedded videos, put them in a <figure>, and mark them for
+        // downloading
+        const videosToProcess = $body.find('.jetpack-video-wrapper')
+              .map(function () {
```
Comment (dnohales) [inline examples/tutorial/index.js]: Too much indentation

File: examples/tutorial/index.js [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+const JSDOM = require('jsdom/lib/old-api');
+const Libingester = require('libingester');
+const url = require('url');
+const util = require('util');
+const { Vimeo } = require('vimeo');
+const Youtubedl = require('youtube-dl');
+
+Youtubedl.getInfo = util.promisify(Youtubedl.getInfo);
+
+const feedURI = 'https://creativecommons.org/blog/feed/';
```
Comment (dnohales) [inline examples/tutorial/index.js]: There is no need for a `const` here.

File: examples/tutorial/index.js (hunk 3 above)
Comment (manuq) [reply examples/tutorial/index.js]: According to? The lint command pass.

File: examples/tutorial/index.js (hunk 4 above)
Comment (manuq) [reply examples/tutorial/index.js]: Like above. Do we need a lint rule?

File: examples/tutorial/index.js (hunk 5 above)
Comment (manuq) [reply examples/tutorial/index.js]: ^

File: examples/tutorial/index.js (hunk 6 above)
Comment (manuq) [reply examples/tutorial/index.js]: ^

File: examples/tutorial/index.js (hunk 7 above)
Comment (manuq) [reply examples/tutorial/index.js]: The original tutorial has it. I think is good to have the feed in a const.

File: examples/tutorial/index.js (hunk 2 above)
Comment (manuq) [reply examples/tutorial/index.js]: Heh, I copy/pasted from the original. Any ideas?

File: examples/tutorial/index.js (hunk 1 above)
Comment (manuq) [reply examples/tutorial/index.js]: +1

File: examples/tutorial/index.js (hunk 2 above)
Comment (dnohales) [reply examples/tutorial/index.js]: Maybe something like:

```
const hasAncestor = (tagName, scoreIfHas) => fnode => {
    const lowerTag = tagName.toLowerCase();

    let element = fnode.element;

    while (element.parentNode !== null && element.parentNode.nodeType === parent.ELEMENT_NODE) {
        if (element.tagName.toLowerCase() === lowerTag) {
            return scoreIfHas;
        }
        element = element.parentNode;
    }

    return 1;
};
```

File: examples/tutorial/index.js (hunk 3 above)
Comment (dnohales) [reply examples/tutorial/index.js]: I don't know, like I feel the indentation should be multiple of 4 in this case.

File: examples/tutorial/index.js (hunk 7 above)
Comment (dnohales) [reply examples/tutorial/index.js]: Well, you declare a `const` only to use it in one place.

File: examples/tutorial/index.js (hunk 2 above)
Comment (manuq) [reply examples/tutorial/index.js]: +1

File: examples/tutorial/index.js (hunk 3 above)
Comment (manuq) [reply examples/tutorial/index.js]: OK I enabled indent MemberExpression in eslint for this.

File: examples/tutorial/index.js (hunk 7 above)
Comment (manuq) [reply examples/tutorial/index.js]: I understand declaring a var or const is useful when you reuse it. But is also good for readability. Honestly I prefer the feed URL at the top in a const, especially in a tutorial.

**Pull request merged**